File: removeDuplicates.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0)
            return 0;
        
        int p1 = 0;
        for (int i = 1; i < nums.length; i++){
            if (nums[p1] != nums[i]){
                p1 += 1;
                nums[p1] = nums[i];
            }
        }
        
        return p1 + 1;
    }
}

File: removeDuplicates.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        
        writeIndex = 1
        
        for i in range(1, len(nums)):
            if nums[writeIndex - 1] != nums[i]:
                nums[writeIndex] = nums[i]
                writeIndex += 1
                
        return writeIndex 